resources/js/register.js

```diff
@@ -17,189 +17,242 @@ let isValid = true
+    setError(field, `${field.name} cannot be blank.`) 
   }
-  else{
+  else {
     if (field.name == 'First name' || field.name == 'Last name') {
-      validateName(field);
-    } 
+      validateName(field) 
+    }
     else if (field.name == 'Password') {
       validatePassword(field)
     }
-    else if(field.name == 'password-confirm'){
+    else if (field.name == 'password-confirm') {
       confirmPassword(field)
     }
+    else if (field.name == 'Date of birth') {
+      validateDate(field)
+    }
   }
 }
 
 // Validate name
 function validateName(input) {
   if (input.value.length < input.minLength) {
-    setError(input, `${input.name} is too short.`);
-    return false;
+    setError(input, `${input.name} is too short.`) 
+    return false 
   } else {
     if (!isName(input.value)) {
-      setError(input, 'Name is invalid!');
-      return false;
+      setError(input, 'Name is invalid!') 
+      return false 
     } else {
-      clearError(input);
-      return true;
+      clearError(input) 
+      return true 
     }
   }
 }
 
 // Validate password
-function validatePassword(input){
-  if(input.value.length < input.minLength){
+function validatePassword(input) {
+  if (input.value.length < input.minLength) {
     setError(input, `${input.name} is too short.`)
     return false
   }
-  else{
-    if(!isPass(input.value)){
+  else {
+    if (!isPass(input.value)) {
       setError(
         input,
         'password should contain at least one number and one special character'
-      );
-      return false;
+      ) 
+      return false 
     }
-    else{
-      clearError(input);
-      return true;
+    else {
+      clearError(input) 
+      return true 
     }
   }
 }
 
 // Confirm password
-function confirmPassword(input){
-  let secondPass = document.querySelector(`#${input.id}-confirm`);
-  if (input.value == secondPass.value){
+function confirmPassword(input) {
+  let secondPass = document.querySelector(`#${input.id}-confirm`) 
+  if (input.value == secondPass.value) {
     return true
   }
-  else{
+  else {
     return false
   }
 }
 
+// Dob validation
+function validateDate(input) {
+  let d = new Date() 
+
+  let dateInput = input.value.split("-")
+  let userYear = parseInt(dateInput[0])
+  let userMonth = parseInt(dateInput[1])
+  let userDay = parseInt(dateInput[2])
+
+  if (userYear > d.getFullYear()) {
+    setError(input, "You are not born yet!")
+  }
+  else if (userYear == d.getFullYear()) {
+    if (userMonth > d.getMonth() + 1 || userMonth > `0${d.getMonth() + 1}`) {
+      setError(input, "You are not born yet!") 
+    }
+    else if (userMonth == d.getMonth() + 1 || userMonth == `0${d.getMonth() + 1}`) {
+      if (userDay >= d.getDate() || userDay >= `0${d.getDate()}`) {
+        setError(input, "You are not born yet!") 
+      }
+      else {
+        clearError(input)
+      }
+    }
+    else {
+      clearError(input)
+    }
+  }
+  else {
+    clearError(input)
+  }
+
```

Comment:
Are these extra validations needed here? Since your logic is to throw error for future dates, isn't the below enough:

```
let d = new Date()
let userDate = new Date(input.value)
if (userDate.getTime() > d.getTime()) {
 setError(input, "Please enter past dates.")
} else {
 clearError(input)
}
```

Pls correct if you are ensuring any other case as well.

Reply:
My lack of knowledge about date property resulted in that twisted logic :). This looks better.

Reply:
(y)